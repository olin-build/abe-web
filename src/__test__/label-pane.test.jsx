import React from 'react';
import renderer from 'react-test-renderer';
import LabelPane from '../components/label-pane';

describe('LabelPane', () => {
  const labels = {
    first: {
      name: 'first',
      id: 'label-1',
      color: 'red',
      description: 'first label',
    },
    second: {
      name: 'second',
      id: 'label-2',
      color: 'green',
      description: 'second label',
      protected: true,
    },
  };
  test('matches baseline', () => {
    const jsx = <LabelPane possibleLabels={labels} />;
    const component = renderer.create(jsx);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('shows selected labels', () => {
    const jsx = <LabelPane possibleLabels={labels} selectedLabels={['first']} />;
    const component = renderer.create(jsx);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('hides unselected labels', () => {
    const jsx = (
      <LabelPane possibleLabels={labels} selectedLabels={['first']} showUnselected={false} />
    );
    const component = renderer.create(jsx);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('disables protected labels', () => {
    const jsx = <LabelPane disableProtectedLabels possibleLabels={labels} />;
    const component = renderer.create(jsx);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('not editable', () => {
    const component = renderer.create(<LabelPane editable={false} possibleLabels={labels} selectedLabels={['first']} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('sorts labels', () => {
    const labels2 = {
      0: { name: 'featured' },
      1: { name: 'a-label-2' },
      2: { name: 'a-label-1' },
      3: { name: 'a-label-4', default: true },
      4: { name: 'a-label-3', default: true },
    };
    // These should come out in order 'featured', 'a-label-3', 'a-label-4', 'a-label-1', a-label-2'.
    // Initially verified from manual inspection of the snapshot.
    Object.values(labels2).forEach((label, ix) => {
      label.id = `id-${ix}`; // eslint-disable-line no-param-reassign
    });
    const component = renderer.create(<LabelPane possibleLabels={labels2} showUnselected />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
