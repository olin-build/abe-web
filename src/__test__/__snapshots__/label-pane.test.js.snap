// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelPane matches snapshot 1`] = `
<div
  className="css-class"
>
  <style
    type="text/css"
  >
    .label.first.button:not(.selected){background-color:white;}
    .label.first:not(.button):not(.selected){border-color:red;color:red;}
    .label.button.first:not(.no-hover):hover,.label.first.selected{background-color:red;}
    .label.second.button:not(.selected){background-color:white;}
    .label.second:not(.button):not(.selected){border-color:green;color:green;}
    .label.button.second:not(.no-hover):hover,.label.second.selected{background-color:green;}
  </style>
  <div
    className="label-selector-list"
  >
    <button
      className="button label first"
      id="label-first"
      onClick={[Function]}
      title="first label"
      type="button"
    >
      <span
        className="ion-pricetag"
      >
         
      </span>
      first
    </button>
    <button
      className="button label second"
      id="label-second"
      onClick={[Function]}
      title="second label"
      type="button"
    >
      <span
        className="ion-pricetag"
      >
         
      </span>
      second
    </button>
  </div>
</div>
`;

exports[`LabelPane not editable 1`] = `
<div
  className="css-class"
>
  <style
    type="text/css"
  >
    .label.first.button:not(.selected){background-color:white;}
    .label.first:not(.button):not(.selected){border-color:red;color:red;}
    .label.button.first:not(.no-hover):hover,.label.first.selected{background-color:red;}
    .label.second.button:not(.selected){background-color:white;}
    .label.second:not(.button):not(.selected){border-color:green;color:green;}
    .label.button.second:not(.no-hover):hover,.label.second.selected{background-color:green;}
  </style>
  <div
    className="label-selector-list"
  >
    <span
      className="label first selected"
      id="label-first"
      title="first label"
    >
      <span
        className="ion-pricetag"
      >
         
      </span>
      first
    </span>
    <span
      className="label second"
      id="label-second"
      title="second label"
    >
      <span
        className="ion-pricetag"
      >
         
      </span>
      second
    </span>
  </div>
</div>
`;

exports[`LabelPane with selected labels 1`] = `
<div
  className="css-class"
>
  <style
    type="text/css"
  >
    .label.first.button:not(.selected){background-color:white;}
    .label.first:not(.button):not(.selected){border-color:red;color:red;}
    .label.button.first:not(.no-hover):hover,.label.first.selected{background-color:red;}
    .label.second.button:not(.selected){background-color:white;}
    .label.second:not(.button):not(.selected){border-color:green;color:green;}
    .label.button.second:not(.no-hover):hover,.label.second.selected{background-color:green;}
  </style>
  <div
    className="label-selector-list"
  >
    <button
      className="button label first selected"
      id="label-first"
      onClick={[Function]}
      title="first label"
      type="button"
    >
      <span
        className="ion-pricetag"
      >
         
      </span>
      first
    </button>
    <button
      className="button label second"
      id="label-second"
      onClick={[Function]}
      title="second label"
      type="button"
    >
      <span
        className="ion-pricetag"
      >
         
      </span>
      second
    </button>
  </div>
</div>
`;
